### ISSUE TITLE
Return proper HTTP errors

### ISSUE THREAD

**Issue by albertpastrana** [intenthq/gitkv#4]
As an API client
I want the service to return proper http errors
so that I know if I have to fix my request or it's that the service is down

*Acceptance Criteria*
This just applies to the code we have for the POC (more may need to be added as the project evolves, to be defined in each new feature/change):

- [x] Repository not found: `404`
- [x] File not found: `404`
- [x] Reference not found: `404`
- [x] No repo/file/reference param: `400`
- [x] Other errors: `500`

**Comment by kittsville** [comment]
As per #24 `reference` will become an optional param, so we'll only need to return `400` for repo/file missing. Although if `${repo}`  was missing it'd just be `localhost:7791/repo/`, which could return a list of the available repos?

Current status (before starting work):
- Repository not found: `404` ✅ 
- File not found: `404` ✅ 
- Reference not found: `404` ✅ 
- No repo/file/reference param: `400`
- Other errors: `500`

**Comment by nathankleyn** [comment]
@kittsville I converted the above list into a task list for you and moved it into the ticket description to make it a little easier to track — this causes it to show like subtasks elsewhere in the UI. 👍 

See https://help.github.com/en/articles/about-task-lists for more info.

**Comment by nathankleyn** [comment]
Re-opening for now as there's still two errors to do above (it got closed automatically by #31).

**Comment by kittsville** [comment]
IIRC those are already done, as they were implemented in #31
